soul-plugin/soul-plugin-ratelimiter/src/main/java/org/dromara/soul/plugin/ratelimiter/executor/RedisRateLimiter.java

```diff
@@ -64,11 +72,15 @@
                     return longs;
                 }).map(results -> {
                     boolean allowed = results.get(0) == 1L;
+                    if (allowed) {
+                        exchange.getAttributes().put(RATE_LIMITER_ALGORITHM, rateLimiterAlgorithm);
+                        exchange.getAttributes().put(KEYS, keys);
+                        exchange.getAttributes().put(SCRIPT_ARGS, scriptArgs);
+                    }
                     Long tokensLeft = results.get(1);
                     return new RateLimiterResponse(allowed, tokensLeft);
                 })
-                .doOnError(throwable -> log.error("Error determining if user allowed from redis:{}", throwable.getMessage()))
-                .doFinally(signalType -> rateLimiterAlgorithm.callback(script, keys, scriptArgs));
+                .doOnError(throwable -> log.error("Error determining if user allowed from redis:{}", throwable.getMessage()));
```

Comment:
It has nothing to do with your changes here, but I think the expression of this diary is 农not very clear.

Reply:
After my test, I found that，RedisRateLimiter#isAllowed “doFinally” execution sequence is :

1.WebClientPlugin#execute send request

2.doFinally execution RateLimiterAlgorithm#callback

3.WebClientResponsePlugin#execute write response

I think a request needs to write response ，then execution RateLimiterAlgorithm#callback, because

Concurrent requests limiter：

“Because Redis is so fast, doing the naive thing works. Just add a random token to a set at the start of a request and remove it from the set when you're done. If the set is too large, reject the request.”

from https://gist.github.com/ptarjan/e38f45f2dfe601419ca3af937fff574d#file-1-check_request_rate_limiter-rb-L11-L34

so the doFinally It should be written in RateLimiterPlugin#doExecute，If so, then execution sequence is :

1.WebClientPlugin#execute send request

2.WebClientResponsePlugin#execute write response

3.doFinally execution RateLimiterAlgorithm#callback

Reply:
I just happened to find that the grammar of this log is not very right. Can you fix it by the way?

Reply:
I found that spring cloud gateway's error log is the same
```java
catch (Exception e) {
	/*
         * We don't want a hard dependency on Redis to allow traffic. Make sure to set
         * an alert so you know if this is happening too much. Stripe's observed
         * failure rate is 0.01%.
         */
	log.error("Error determining if user allowed from redis", e);
}
```
https://github.com/spring-cloud/spring-cloud-gateway/blob/main/spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/ratelimit/RedisRateLimiter.java

line 284